<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SymmetryPlaneFromPoints</class>
 <widget class="qMRMLWidget" name="SymmetryPlaneFromPoints">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>450</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Symmetry Plane From Points</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="inputsCollapsibleButton">
     <property name="text">
      <string>Select Points</string>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="point1Label">
          <property name="text">
           <string>Point 1 Label:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="point1LineEdit">
          <property name="toolTip">
           <string>Enter the label of first point (e.g., S, N, Ba, Me)</string>
          </property>
          <property name="placeholderText">
           <string>e.g., S or N</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="point2Label">
          <property name="text">
           <string>Point 2 Label:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="point2LineEdit">
          <property name="toolTip">
           <string>Enter the label of second point</string>
          </property>
          <property name="placeholderText">
           <string>e.g., N or Ba</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="point3Label">
          <property name="text">
           <string>Point 3 Label:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="point3LineEdit">
          <property name="toolTip">
           <string>Enter the label of third point</string>
          </property>
          <property name="placeholderText">
           <string>e.g., Ba or Me</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="point4Label">
          <property name="text">
           <string>Point 4 Label (optional):</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="point4LineEdit">
          <property name="toolTip">
           <string>Optional: Enter the label of fourth point for best-fit plane</string>
          </property>
          <property name="placeholderText">
           <string>e.g., Me (optional)</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="instructionLabel">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt; font-style:italic;&quot;&gt;Enter the labels of existing fiducial points (e.g., S, N, Ba, Me).&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="outputsCollapsibleButton">
     <property name="text">
      <string>Create Plane</string>
     </property>
     <property name="collapsed">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="statusLabel">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:9pt;&quot;&gt;Status: &lt;/span&gt;&lt;span style=&quot; font-size:9pt; font-weight:600; color:#ff0000;&quot;&gt;Need at least 3 points&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="createPlaneButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Create symmetry plane from selected points</string>
        </property>
        <property name="text">
         <string>Create Plane</string>
        </property>
        <property name="icon">
         <iconset>
          <normaloff>:/Icons/SlicesCrosshair.png</normaloff>:/Icons/SlicesCrosshair.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>24</width>
          <height>24</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="outputInfoLabel">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:8pt; font-style:italic;&quot;&gt;The plane will be created as a model node named &amp;quot;SymmetryPlane&amp;quot; with light blue color and 40% opacity.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  <item>
   <widget class="ctkCollapsibleButton" name="sheetCollapsibleButton">
    <property name="text">
     <string>Cutting Sheet</string>
    </property>
    <property name="collapsed">
     <bool>true</bool>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_sheet">
     <item>
      <layout class="QFormLayout" name="formLayoutSheet">
       <item row="0" column="0">
        <widget class="QLabel" name="curveNameLabel">
         <property name="text"><string>Curve name:</string></property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="curveNameLineEdit">
         <property name="placeholderText"><string>MOL</string></property>
         <property name="toolTip"><string>Name of Markups Curve node (default MOL)</string></property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="yawLabel">
         <property name="text"><string>Yaw (deg):</string></property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="yawSpinBox">
         <property name="minimum"><double>-90.0</double></property>
         <property name="maximum"><double>90.0</double></property>
         <property name="value"><double>45.0</double></property>
         <property name="singleStep"><double>1.0</double></property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="latLabel">
         <property name="text"><string>Lateral out (mm):</string></property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="latSpinBox">
         <property name="minimum"><double>0.0</double></property>
         <property name="maximum"><double>50.0</double></property>
         <property name="value"><double>5.0</double></property>
         <property name="singleStep"><double>0.5</double></property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="medLabel">
         <property name="text"><string>Medial in (mm):</string></property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QDoubleSpinBox" name="medSpinBox">
         <property name="minimum"><double>0.0</double></property>
         <property name="maximum"><double>80.0</double></property>
         <property name="value"><double>15.0</double></property>
         <property name="singleStep"><double>0.5</double></property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="extendLabel">
         <property name="text"><string>Extend ends (mm):</string></property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QDoubleSpinBox" name="extendSpinBox">
         <property name="minimum"><double>0.0</double></property>
         <property name="maximum"><double>40.0</double></property>
         <property name="value"><double>10.0</double></property>
         <property name="singleStep"><double>1.0</double></property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="createSheetButton">
       <property name="text"><string>Create Cutting Sheet</string></property>
       <property name="toolTip"><string>Build sheet from curve and mirror across MSP</string></property>
      </widget>
     </item>
  <item>
   <widget class="QLabel" name="sheetStatusLabel">
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:8pt; font-style:italic;&quot;&gt;Requires an MSP plane and a Markups Curve (e.g. MOL).&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="wordWrap"><bool>true</bool></property>
   </widget>
  </item>
    </layout>
   </widget>
  </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLWidget</class>
   <extends>QWidget</extends>
   <header>qMRMLWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
